lua/cinema/scoreboard/cl_playerlist.lua

```diff
@@ -51,11 +51,28 @@ concommand.Add("mute", function(ply, cmd, args, argss)
 
     if v then
         v.ClickMuted = not v.ClickMuted
-        print(v.ClickMuted and "Muted" or "Unmuted", v)
-        UpdateMutes()
+        UpdateMuteHistory(v)
+        print(v.ClickMuted and " Muted" or " Unmuted", v)
     end
 end)
 
+local history = util.JSONToTable(file.Read("swamp_mutehistory.txt", "DATA") or "") or {}
+function UpdateMuteHistory(ply)
+    if ply then
+        history[ply:SteamID()] = (ply.ClickMuted or ply.IsChatMuted) and {mute=ply.ClickMuted, chatmute=ply.IsChatMuted} or nil
+    end
+
+    for k,v in pairs(history) do
+        local p = player.GetBySteamID(k)
```

Comment:
You have an `O(numplayers * historysize)` function here and it runs every time a new player gets created. Factor your code properly so you're only changing the ClickMuted/IsChatMuted/SetMuted/whatever on individual players instead of resetting them all and rewriting your file every time a player enters PVS. Do not repeatedly set things if there is no reason to.